src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/credentials/BitbucketOAuthCredentialMatcher.java

```diff
@@ -19,13 +22,18 @@ public boolean matches(Credentials item) {
         if (!(item instanceof UsernamePasswordCredentials))
             return false;
 
-        UsernamePasswordCredentials usernamePasswordCredential = ((UsernamePasswordCredentials) item);
-        String username = usernamePasswordCredential.getUsername();
-        boolean isEMail = username.contains(".") && username.contains("@");
-        boolean validSecretLenght = Secret.toString(usernamePasswordCredential.getPassword()).length() == secretLenght;
-        boolean validKeyLenght = username.length() == keyLenght;
-
-        return !isEMail && validKeyLenght && validSecretLenght;
+        try {
+            UsernamePasswordCredentials usernamePasswordCredential = ((UsernamePasswordCredentials) item);
+            String username = usernamePasswordCredential.getUsername();
+            boolean isEMail = username.contains(".") && username.contains("@");
+            boolean validSecretLenght = Secret.toString(usernamePasswordCredential.getPassword()).length() == secretLenght;
+            boolean validKeyLenght = username.length() == keyLenght;
+
+            return !isEMail && validKeyLenght && validSecretLenght;
+        } catch(Exception e) {
```

Comment:
Seems a little broad.

Doesn't really seem like a fix. More like your just logging the exception.

Reply:
The idea here is to be able to continue, if some of the defined credentials can't get the password

Reply:
Than why log the exception at all?

I feel like we would be better suited creating a new credential class.

Reply:
That way, if the throwing credential should contains the OAuth key/secret then the user have a way to know what happened.

But I do agree for the new credential class, but I really don't know how to do that.

Reply:
Here is a few examples: https://github.com/search?q=org%3Ajenkinsci+extends+BaseStandardCredentials&type=Code

Reply:
Here are the docs: https://github.com/jenkinsci/credentials-plugin/blob/master/docs/implementation.adoc#implementing-a-new-credentials-type

Reply:
@kurddt A new credential type was also created recently in the github-branch-source-plugin. You can look there for a concrete example.